shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/cache/ResponseCache.kt

```diff
@@ -2,67 +2,87 @@ package com.mbta.tid.mbta_app.cache
 
 import io.ktor.client.plugins.ResponseException
 import io.ktor.client.statement.HttpResponse
+import io.ktor.client.statement.bodyAsText
 import io.ktor.http.HttpStatusCode
+import io.ktor.http.etag
 import kotlin.time.Duration
 import kotlin.time.Duration.Companion.hours
 import kotlin.time.TimeSource
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
 import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import org.koin.core.parameter.parametersOf
 
-sealed class ConditionalResponse {
-    data object NotModified : ConditionalResponse()
+@Serializable
+data class Response(
+    val etag: String?,
+    @Serializable(with = TimeMarkSerializer::class)
+    var fetchTime: TimeSource.Monotonic.ValueTimeMark,
+    val data: String
+)
 
-    data class Response(val etag: String?, val response: HttpResponse) : ConditionalResponse()
-}
-
-class ResponseCache(val maxAge: Duration = 1.hours) : KoinComponent {
-    internal var data: ConditionalResponse.Response? = null
-    internal var dataTimestamp: TimeSource.Monotonic.ValueTimeMark? = null
+class ResponseCache(private val cacheKey: String, val maxAge: Duration = 1.hours) : KoinComponent {
+    internal var data: Response? = null
+    internal val cacheFile: CacheFile by inject { parametersOf(cacheKey) }
     private val lock = Mutex()
 
-    private fun getData(): HttpResponse? {
-        val data = this.data ?: return null
-        val dataTimestamp = this.dataTimestamp ?: return null
-        return if (dataTimestamp.elapsedNow() < maxAge) {
-            data.response
+    private fun getData(): Response? {
+        val data = this.data ?: readData() ?: return null
+        return if (data.fetchTime.elapsedNow() < maxAge) {
+            data
         } else {
             null
```

Comment:
Right now we don't actually ever delete cached data once it's written to disk. The obvious place for deletes would be right here when the data is stale, but it's then possible to get a NotModified response right after, and it seems like a waste to be constantly deleting and recreating the file every time we get NotModified.